packages-ui/roosterjs-react/lib/emoji/plugin/createEmojiPlugin.ts

```diff
@@ -0,0 +1,426 @@
+        }
+
+        this.strings = {
+            ...EmojiDescriptionStrings,
+            ...EmojiKeywordStrings,
+            ...EmojiFamilyStrings,
+        };
+    }
+
+    setUIUtilities(uiUtilities: UIUtilities) {
+        this.uiUtilities = uiUtilities;
+    }
+
+    willHandleEventExclusively?: (event: PluginEvent) => boolean;
+
+    public getName() {
+        return 'Emoji';
+    }
+
+    public dispose() {
+        this.setIsSuggesting(false);
+        if (this.contentDiv) {
+            ReactDOM.unmountComponentAtNode(this.contentDiv);
+            this.contentDiv.parentElement.removeChild(this.contentDiv);
+            this.contentDiv = null;
+        }
+        this.editor = null;
+    }
+
+    public initialize(editor: Editor): void {
+        this.editor = editor;
+        const document = editor.getDocument();
+        this.contentDiv = document.createElement('div');
+        document.body.appendChild(this.contentDiv);
+    }
+
+    public initializeContentEditable(contentEditable: HTMLDivElement): void {
+        this.contentEditable = contentEditable;
+    }
+
+    public onPluginEvent(event: PluginEvent): void {
+        const domEvent = event as PluginDomEvent;
+        const keyboardEvent = domEvent.rawEvent as KeyboardEvent;
+        if (event.eventType === PluginEventType.KeyDown) {
+            this.eventHandledOnKeyDown = false;
+            if (this.isSuggesting) {
+                this.onKeyDownSuggestingDomEvent(domEvent);
+            } else if (keyboardEvent.which === KeyCodes.backspace && this.canUndoEmoji) {
+                // If KeyDown is backspace and canUndoEmoji, call editor undo
+                this.editor.undo();
+                this.handleEventOnKeyDown(domEvent);
+                this.canUndoEmoji = false;
+            }
+        } else if (
+            event.eventType === PluginEventType.KeyUp &&
+            !this.isModifierKey(keyboardEvent.key)
+        ) {
+            if (this.isSuggesting) {
+                this.onKeyUpSuggestingDomEvent(domEvent);
+            } else {
+                this.onKeyUpDomEvent(domEvent);
+            }
+        } else if (event.eventType === PluginEventType.MouseUp) {
+            // If MouseUp, the emoji cannot be undone
+            this.canUndoEmoji = false;
+            this.setIsSuggesting(false);
+        }
+    }
+
+    /**
+     * On KeyDown suggesting DOM event
+     * Try to insert emoji is possible
+     * Intercept arrow keys to move selection if popup is shown
+     */
+    private onKeyDownSuggestingDomEvent(event: PluginDomEvent): void {
+        // If key is enter, try insert emoji at selection
+        // If key is space and selection is shortcut, try insert emoji
+        const keyboardEvent = event.rawEvent as KeyboardEvent;
+        const selectedEmoji = this.pane.getSelectedEmoji();
+        const wordBeforeCursor = this.getWordBeforeCursor(event);
+
+        let emoji: Emoji;
+        switch (keyboardEvent.which) {
```

Comment:
Open a bug to port this plugin to use PickerPlugin, and add bug number with "// TODO" here